internal/provider/repository_branch_protection_resource.go

```diff
@@ -0,0 +1,816 @@
+	_ resource.ResourceWithImportState = &repositoryBranchProtectionResource{}
+)
+
+type repositoryBranchProtectionResource struct {
+	client *forgejo.Client
+}
+
+// repositoryBranchProtectionResourceModel maps the resource schema data.
+// https://pkg.go.dev/codeberg.org/mvdkleijn/forgejo-sdk/forgejo/v2#CreateBranchProtectionOption
+type repositoryBranchProtectionResourceModel struct {
+	ID                            types.String `tfsdk:"id"`
+	Owner                         types.String `tfsdk:"owner"`
+	Repo                          types.String `tfsdk:"repo"`
+	BranchName                    types.String `tfsdk:"branch_name"`
+	EnablePush                    types.Bool   `tfsdk:"enable_push"`
+	EnablePushWhitelist           types.Bool   `tfsdk:"enable_push_whitelist"`
+	PushWhitelistUsernames        types.List   `tfsdk:"push_whitelist_usernames"`
+	PushWhitelistTeams            types.List   `tfsdk:"push_whitelist_teams"`
+	PushWhitelistDeployKeys       types.Bool   `tfsdk:"push_whitelist_deploy_keys"`
+	EnableStatusCheck             types.Bool   `tfsdk:"enable_status_check"`
+	StatusCheckContexts           types.List   `tfsdk:"status_check_contexts"`
+	RequireSignedCommits          types.Bool   `tfsdk:"require_signed_commits"`
+	ProtectedFilePatterns         types.String `tfsdk:"protected_file_patterns"`
+	UnprotectedFilePatterns       types.String `tfsdk:"unprotected_file_patterns"`
+	EnableMergeWhitelist          types.Bool   `tfsdk:"enable_merge_whitelist"`
+	MergeWhitelistUsernames       types.List   `tfsdk:"merge_whitelist_usernames"`
+	MergeWhitelistTeams           types.List   `tfsdk:"merge_whitelist_teams"`
+	EnableApprovalsWhitelist      types.Bool   `tfsdk:"enable_approvals_whitelist"`
+	ApprovalsWhitelistUsernames   types.List   `tfsdk:"approvals_whitelist_usernames"`
+	ApprovalsWhitelistTeams       types.List   `tfsdk:"approvals_whitelist_teams"`
+	RequiredApprovals             types.Int64  `tfsdk:"required_approvals"`
+	BlockOnRejectedReviews        types.Bool   `tfsdk:"block_on_rejected_reviews"`
+	BlockOnOfficialReviewRequests types.Bool   `tfsdk:"block_on_official_review_requests"`
+	BlockOnOutdatedBranch         types.Bool   `tfsdk:"block_on_outdated_branch"`
+	DismissStaleApprovals         types.Bool   `tfsdk:"dismiss_stale_approvals"`
+}
+
+func NewRepositoryBranchProtectionResource() resource.Resource {
+	return &repositoryBranchProtectionResource{}
+}
+
+func (r *repositoryBranchProtectionResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_repository_branch_protection"
+}
+
+func (r *repositoryBranchProtectionResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Forgejo repository branch protection resource.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
```

Comment:
Is this ID used anywhere? SDKv2 required all resources to have an ID attribute, but the Plugin Framework doesn't have this limitation. Hence, no need to construct an "artificial" ID ourselves...